File: src/contests/cg/puzzle/swift/sum-of-divisors.swift
// 🎮 CodinGame Puzzle - sum-of-divisors
// https://www.codingame.com/training/medium/sum-of-divisors

// d appears as a divisor in floor(n/d) of the numbers 1..n, so the total sum
// of divisors is sum over d of d * floor(n/d).
let n = Int(readLine()!)!
var total = 0
var d = 1
while d <= n {
    total += d * (n / d)
    d += 1
}
print(total)

File: src/contests/cg/puzzle/csharp/sum-of-divisors.cs
// 🎮 CodinGame Puzzle - sum-of-divisors
// https://www.codingame.com/training/medium/sum-of-divisors

using System;

class Solution
{
    static void Main()
    {
        // d appears as a divisor in floor(n/d) of the numbers 1..n, so the total
        // sum of divisors is sum over d of d * floor(n/d).
        long n = long.Parse(Console.ReadLine().Trim());
        long total = 0;
        for (long d = 1; d <= n; d++) total += d * (n / d);
        Console.WriteLine(total);
    }
}
